Stop using unsafeMakeSpan() in BinaryPropertyList.cpp

cdumez opened the pull request:
#### 8f4d4da6c5ed0570669f64ff44b52bc84e44b843
<pre>
Stop using unsafeMakeSpan() in BinaryPropertyList.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=291772">https://bugs.webkit.org/show_bug.cgi?id=291772</a>

Reviewed by Geoffrey Garen, Darin Adler, and Brady Eidson.

* Source/WebKitLegacy/mac/History/BinaryPropertyList.cpp:
(BinaryPropertyListSerializer::BinaryPropertyListSerializer):
* Source/WebKitLegacy/mac/History/BinaryPropertyList.h:
* Source/WebKitLegacy/mac/History/HistoryPropertyList.h:
* Source/WebKitLegacy/mac/History/HistoryPropertyList.mm:
(CFMalloc::malloc):
(CFMalloc::free):
(HistoryPropertyListWriter::HistoryPropertyListWriter):
(HistoryPropertyListWriter::buffer):
(HistoryPropertyListWriter::releaseData):

Canonical link: <a href="https://commits.webkit.org/294197@main">https://commits.webkit.org/294197@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/80b967f4c9e085f33cdd00b6bbeb6885a96b3709

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101121 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/20783 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11086 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106269 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/51748 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21092 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/29277 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77028 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34054 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104128 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16249 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91332 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57375 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16066 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/9361 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51096 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/85971 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9415 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/108625 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28249 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/20802 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/85996 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/28611 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/87533 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/85533 "Found 101 new API test failures: /TestWebKit:WebKit.EnumerateDevicesCrash, /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/autoplay-policy, /WebKitGTK/TestUIClient:/webkit/WebKitWebView/file-chooser-request, /TestWebKit:WebKit.PageLoadDidChangeLocationWithinPage, /WebKitGTK/TestOptionMenu:/webkit/WebKitWebView/option-menu-activate, /WebKitGTK/TestInputMethodContext:/webkit/WebKitInputMethodContext/simple, /WebKitGTK/TestUIClient:/webkit/WebKitWebView/display-usermedia-permission-request, /WebKitGTK/TestInspectorServer:/webkit/WebKitWebInspectorServer/test-page-list, /WebKitGTK/TestUIClient:/webkit/WebKitWebView/open-window-no-default-size, /TestWebKit:WebKit.LoadPageAfterCrash ... (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21763 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30259 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/7975 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/22347 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28179 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/33448 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/27991 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31311 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/29549 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebKitLegacy/mac/History/HistoryPropertyList.mm:
```diff
@@ -32,36 +32,38 @@
+    if (!m_buffer)
+        return nullptr;
+    auto span = m_buffer.leakSpan();
+    RetainPtr data = adoptCF(CFDataCreateWithBytesNoCopy(0, span.data(), span.size(), 0));
```
This is not new to your patch, but how does the CFData end up freeing the span it adopts? I would imagine that BytesNoCopy prevents calling the deallocator.

darinadler replied:
No, the deallocator is called even if we use this create function. If we didn't want it to deallocate we'd have to pass `kCFAllocatorNull`. Here we pass `kCFAllocatorDefault`, although we write it `0`.

darinadler replied:
A special thing about the `NoCopy` function is that the caller passes a separate deallocator for the bytes, the last argument to the function, whereas the other functions use the same allocator that is used to allocate the structure itself, the first argument to the function, for both allocating and deallocating the bytes.

darinadler commented on Source/WebKitLegacy/mac/History/HistoryPropertyList.h:
```diff
@@ -54,8 +57,8 @@ class HistoryPropertyListWriter : public BinaryPropertyListWriter {
 
-    UInt8* m_buffer;
-    size_t m_bufferSize;
+
```
Extra blank line here.

cdumez committed:
Stop using unsafeMakeSpan() in BinaryPropertyList.cpp
https://bugs.webkit.org/show_bug.cgi?id=291772

Reviewed by Geoffrey Garen, Darin Adler, and Brady Eidson.

* Source/WebKitLegacy/mac/History/BinaryPropertyList.cpp:
(BinaryPropertyListSerializer::BinaryPropertyListSerializer):
* Source/WebKitLegacy/mac/History/BinaryPropertyList.h:
* Source/WebKitLegacy/mac/History/HistoryPropertyList.h:
* Source/WebKitLegacy/mac/History/HistoryPropertyList.mm:
(CFMalloc::malloc):
(CFMalloc::free):
(HistoryPropertyListWriter::HistoryPropertyListWriter):
(HistoryPropertyListWriter::buffer):
(HistoryPropertyListWriter::releaseData):

Canonical link: https://commits.webkit.org/294197@main
